### PR TITLE
core_rpc_server: on_get_blocks - forward bootstrap daemon error

### PR DESCRIPTION
**Pull request by xiphon**

### CODE REVIEW

**Review by selsta — APPROVED** [review]

**Review by selsta — APPROVED** [review]

File: src/rpc/core_rpc_server.cpp [hunk 1]
```diff
@@ -521,9 +521,15 @@ namespace cryptonote
   bool core_rpc_server::on_get_blocks(const COMMAND_RPC_GET_BLOCKS_FAST::request& req, COMMAND_RPC_GET_BLOCKS_FAST::response& res, const connection_context *ctx)
   {
     RPC_TRACKER(get_blocks);
-    bool r;
-    if (use_bootstrap_daemon_if_necessary<COMMAND_RPC_GET_BLOCKS_FAST>(invoke_http_mode::BIN, "/getblocks.bin", req, res, r))
-      return r;
+
+    {
+      boost::shared_lock<boost::shared_mutex> lock(m_bootstrap_daemon_mutex);
```
Comment (moneromooo-monero) [inline src/rpc/core_rpc_server.cpp:526]: That doesn't look like a recursive mutex (though some platforms might make it recursive) and use_bootstrap_daemon_if_necessary takes it too. That might lockup.

Comment (xiphon) [reply src/rpc/core_rpc_server.cpp:527]: Yep, fixed. Introduced a race condition instead, but i consider this one is acceptable.

**Review by selsta — APPROVED** [review]
tested

**Review by moneromooo-monero — APPROVED** [review]
Yes, the race  makes it do what it does now, so I agree it's acceptable

**Pull request merged**